src/contracts/pools/FixedRateExchange.ts

```diff
@@ -0,0 +1,773 @@
+import FixedRateExchangeAbi from '@oceanprotocol/contracts/artifacts/contracts/pools/fixedRate/FixedRateExchange.sol/FixedRateExchange.json'
+import { TransactionReceipt } from 'web3-core'
+import { AbiItem } from 'web3-utils/types'
+import { calculateEstimatedGas, ZERO_ADDRESS } from '../../utils'
+import { PriceAndFees, FeesInfo, FixedPriceExchange } from '../../@types'
+import { SmartContractWithAddress } from '..'
+
+export class FixedRateExchange extends SmartContractWithAddress {
+  getDefaultAbi(): AbiItem | AbiItem[] {
+    return FixedRateExchangeAbi.abi as AbiItem[]
+  }
+
+  /**
+   * Creates unique exchange identifier.
+   * @param {String} baseToken baseToken contract address
+   * @param {String} datatoken Datatoken contract address
+   * @return {Promise<string>} exchangeId
+   */
+  public async generateExchangeId(baseToken: string, datatoken: string): Promise<string> {
+    const exchangeId = await this.contract.methods
+      .generateExchangeId(baseToken, datatoken)
+      .call()
+    return exchangeId
+  }
+
+  /**
+   * Atomic swap
+   * @param {String} exchangeId ExchangeId
+   * @param {String} datatokenAmount Amount of datatokens
+   * @param {String} maxBaseTokenAmount max amount of baseToken we want to pay for datatokenAmount
+   * @param {String} address User address
+   * @param {String} consumeMarketAddress consumeMarketAddress
+   * @param {String} consumeMarketFee consumeMarketFee in fraction
+   * @return {Promise<TransactionReceipt>} transaction receipt
+   */
+  public async buyDatatokens<G extends boolean = false>(
+    address: string,
+    exchangeId: string,
+    datatokenAmount: string,
+    maxBaseTokenAmount: string,
+    consumeMarketAddress: string = ZERO_ADDRESS,
+    consumeMarketFee: string = '0',
+    estimateGas?: G
+  ): Promise<G extends false ? TransactionReceipt : number> {
+    const exchange = await this.getExchange(exchangeId)
+    const consumeMarketFeeFormatted = this.web3.utils.toWei(consumeMarketFee)
+    const dtAmountFormatted = await this.amountToUnits(
+      exchange.datatoken,
+      datatokenAmount,
+      +exchange.dtDecimals
+    )
+    const maxBtFormatted = await this.amountToUnits(
+      exchange.baseToken,
+      maxBaseTokenAmount,
+      +exchange.btDecimals
+    )
+
+    const estGas = await calculateEstimatedGas(
+      address,
+      this.contract.methods.buyDT,
+      exchangeId,
+      dtAmountFormatted,
+      maxBtFormatted,
+      consumeMarketAddress,
+      consumeMarketFeeFormatted
+    )
+    if (estimateGas) return estGas
+
+    const trxReceipt = await this.contract.methods
```

Comment:
not sure why the try catch clause was removed from here , maybe the error logs would be helpful